packages/phone-number-privacy/signer/src/common/database/wrappers/domainState.ts

```diff
@@ -102,33 +94,32 @@ export async function updateDomainStateRecord<D extends Domain>(
   trx: Knex.Transaction<DomainStateRecord>,
   logger: Logger
 ): Promise<void> {
-  const meter = Histograms.dbOpsInstrumentation.labels('updateDomainStateRecord').startTimer()
-  const hash = domainHash(domain).toString('hex')
-  logger.debug({ hash, domain, domainState }, 'Update domain state')
-  try {
-    // Check whether the domain is already in the database.
-    // TODO(2.0.0, refactor): Usage of this in the signature flow results in redudant queries of the current
-    // state. It would be good to refactor this to avoid making more than one SELECT.
-    // Consider doing this as part of DB audit ticket (https://github.com/celo-org/celo-monorepo/issues/9795)
-    const result = await getDomainStateRecord(db, domain, logger, trx)
+  return meter(
+    async () => {
+      const hash = domainHash(domain).toString('hex')
+      logger.debug({ hash, domain, domainState }, 'Update domain state')
+      // Check whether the domain is already in the database.
+      // TODO(2.0.0, refactor): Usage of this in the signature flow results in redudant queries of the current
+      // state. It would be good to refactor this to avoid making more than one SELECT.
+      // Consider doing this as part of DB audit ticket (https://github.com/celo-org/celo-monorepo/issues/9795)
```

Comment:
The only ways I could come up with to avoid this second select would be moving the nonce check -> `checkAndUpdateQuotaStatus` or else checking against the value of `createEmptyOrReturn`, both of which I think are a bit messy & introduce more confusion/complexity so opted to keep as is for now. I'd recommend we remove this TODO in favor of a longer-term ticket that we deprioritize for 2.0.0.

Reply:
Sounds good to me. Thanks for investigating!

Reply:
updated the TODO & linked a ticket for the future :)